lib/azure/armrest/network/virtual_network_service.rb

```diff
@@ -0,0 +1,115 @@
+module Azure
+  module Armrest
+    module Network
+      # Class for managing virtual networks.
+      class VirtualNetworkService < ArmrestService
+
+        # Creates and returns a new VirtualNetworkService instance.
+        #
+        def initialize(_armrest_configuration, options = {})
+          super
+          @provider = options[:provider] || 'Microsoft.Network'
+          set_service_api_version(options, 'virtualNetworks')
+        end
+
+        # Return information for the given virtual network for the provided
+        # +resource_group+. If no group is specified, it will use the resource
+        # group set in the constructor.
+        #
+        # Example:
+        #
+        #   vns.get('vn_name', 'your_resource_group')
+        #
+        def get(vn_name, resource_group = armrest_configuration.resource_group)
+          raise ArgumentError, "must specify resource group" unless resource_group
+          url = build_url(resource_group, vn_name)
+          JSON.parse(rest_get(url))
+        end
+
+        # Returns a list of available virtual networks in the current subscription
+        # for the provided +resource_group+.
+        #
+        def list(resource_group = armrest_configuration.resource_group)
```

Comment:
Because I thought we were going to refactor the other classes to remove the threaded code anyway.